asmcomp/cmm.mli

```diff
@@ -159,8 +159,11 @@ and expression =
   | Cblockheader of nativeint * Debuginfo.t
   | Cvar of Backend_var.t
   | Clet of Backend_var.With_provenance.t * expression * expression
+  | Clet_mut of Backend_var.With_provenance.t * machtype
+                * expression * expression
   | Cphantom_let of Backend_var.With_provenance.t
       * phantom_defining_expr option * expression
+  (* Cassign must refer to a variable bound by Clet_mut *)
```

Comment:
It should be possible to enforce this with types, or with checks. Although I believe at the moment this always holds.

Reply:
Good idea. I'm adding a check. (Changing the types used for names here would be good, but is a big change as it would need to propagate quite far back in the compiler)

Reply:
Check is added (see the second commit here). Turns out there was one place where this didn't hold: the asmgen tests.